tests/api/test_cluster.cpp

```diff
@@ -0,0 +1,76 @@
+
+#include <gtest/gtest.h>
+#include "fmt/xchar.h"
+
+#include <algorithm>
+#include <filesystem>
+#include <string>
+#include <vector>
+
+#include "tests/test_utils/generate_cluster_desc.hpp"
+
+// TODO: change to tt_cluster
+#include "device/tt_device.h"
+#include "device/tt_cluster_descriptor.h"
+
+// TODO: do proper renaming.
+using Cluster = tt_SiliconDevice;
+
+// These tests are intended to be run with the same code on all kinds of systems:
+// E75, E150, E300
+// N150. N300
+// Galaxy
+
+// This test should be one line only.
+TEST(ApiTest, OpenAllChips) {
+
+    // TODO: This should not be needed. And could be part of the cluster descriptor probably.
+    // Note that cluster descriptor holds logical ids of chips.
+    // Which are different than physical PCI ids, which are /dev/tenstorrent/N ones.
+    // You have to see if physical PCIe is GS before constructing a cluster descriptor.
+    std::vector<int> pci_device_ids = PCIDevice::enumerate_devices();
+    std::set<int> pci_device_ids_set;
```

Comment:
nit: You can avoid the loop like, `std::set<int> s{ pci_device_ids.being(), pci_device_ids.end() }`
Although maybe having the cluster accept a vector rather than a set makes more sense?

Reply:
Will change it.
I won't change the API as part of this PR (changing it to accept a vector), I wrote this test mainly to just to document our current API usage.